### ISSUE TITLE
Error shown when valid intrinsic functions used within DeletionPolicy & UpdateReplacePolicy

### ISSUE THREAD

**Issue by arthurboghossian** [aws-cloudformation/cfn-lint#2915]
### CloudFormation Lint Version

cfn-lint 0.80.2

### What operating system are you using?

Mac

### Describe the bug

Receive `[cfn-lint] E3035: DeletionPolicy only supports one of the Fn::FindInMap, Fn::If, Ref intrinsic functions for Resources/Topic/DeletionPolicy` when using valid intrinsic functions that are supported like `Fn::Join`, `Fn::Select`, `Fn::Sub`.

Also see no cfn-lint message when using invalid intrinsic functions nested within the DeletionPolicy & UpdateReplacePolicy attributes.

### Expected behavior

cfn-lint shouldn't give an error message when valid intrinsic functions like `Fn::Join`, and `Fn::Select`, and `Fn::Sub` are used.

Note: if the `Fn::GetAtt`, `Fn::GetAZs`, `Fn::ImportValue` intrinsic functions (or `Ref` to non-parameter and non-AWS pseudo-parameter) are included within the DeletionPolicy (including as a nested intrinsic function), the template is invalid and ideally this would be mentioned by cfn-lint.

### Reproduction template

```yaml
AWSTemplateFormatVersion: 2010-09-09
Parameters:
  Suffix:
    Type: String
    Default: 'elete'
Resources:
  Topic:
    DeletionPolicy: !Sub 'D${Suffix}'
    Type: AWS::SNS::Topic
```

**Comment by kddejong** [comment]
Validated the provided template worked.  Working on our side to understand the docs [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/function-refs-in-policy-attributes.html#function-refs-in-policy-attributes-supported-functions) in relation to this template working with other functions and without the language transformation.

**Comment by kddejong** [comment]
This is indeed possible with the following caveats:
- The intrinsic function must resolve to a valid String option (see options for DeletionPolicy [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-deletionpolicy.html#aws-attribute-deletionpolicy-options)) (Ex: something like Fn::Split, which resolves to a list isn't allowed)
- The intrinsic function must be resolved during pre-processing (Ex: intrinsic functions like Fn::GetAtt, Fn::GetAZs, and Fn::ImportValue, aren't able to be specified as part of the DeletionPolicy/UpdateReplacePolicy)

Documentation is being updated and this issue will track the fixes for cfn-lint

**Comment by kddejong** [comment]
#2928

**Comment by kddejong** [comment]
Added more functions into support.  Released in v0.83.1
